master/internal/api_config_policies.go

```diff
@@ -164,7 +315,12 @@ func (a *apiServer) PutGlobalConfigPolicies(
 		return nil, err
 	}
 
-	err = validatePoliciesAndWorkloadType(req.WorkloadType, req.ConfigPolicies)
+	globalConfigPolicies, err := configpolicy.GetTaskConfigPolicies(ctx, nil, req.WorkloadType)
```

Comment:
Why fetch the old global config policies here? Config policies are completely replaced with every update. To put it another way, the API receives the entire set of config policies with each PUT request. 

I think we should be able to pass in `nil` for global config policies in `validatePoliciesAnd...`

Reply:
Then how will we address this part of the ticket description?
```
if an admin is setting workspace task config policies, but there’s already a global task config policy for that same field

e.g., constraints.resources.max_slots is set at the global level and a workspace admin adds the same constraint for a workspace

also applies to invariant_configs
```

Reply:
This is the Global API; it is only used to set config policies at the global level. There's no need to check old global policies against new ones. Just any existing global policies with any new workspace config policies.

Reply:
how do I get global config policies then?

Reply:
The only global config policies needed in `PutGlobalConfigPolicies` are the ones in the API request: `PutWorkspaceConfigPoliciesRequest.ConfigPolicies`. 

I can sync on this in a bit. Might be easier to chat about it live?